### DISCUSSION TITLE
Lost KeyCloak password of admin user of master realm on Kubernetes/docker setup

### DISCUSSION THREAD

**Discussion by Nickk4** [keycloak/keycloak#20903] [Q&A]
Someone installed KeyCloak (21.0.1) for me on a server with a Kubernetes/docker setup (Digital Ocean). 
However, I have lost my password to the master realm and I'm the only admin user. I also don't know the client secret.
How to solve this situation?

### What can I still do?
- Access the KeyCloak Postgres database. Here I see there are 4 users, including the 1 admin user for whom I've lost the password.
- I can log in on the container on which KeyCloak is installed.

### What have I tried
![image](https://github.com/keycloak/keycloak/assets/10728166/33b45ed9-84fb-4ddd-9140-9b0dadfd2195)

![image](https://github.com/keycloak/keycloak/assets/10728166/3aa971d3-31d0-4a0c-acac-d4ec65565c4d)

![image](https://github.com/keycloak/keycloak/assets/10728166/bfe5a29d-539b-41df-9532-73f5d1552e0f)

![image](https://github.com/keycloak/keycloak/assets/10728166/3e6c40b6-6143-4d7d-ba9d-4d90f3146734)

![image](https://github.com/keycloak/keycloak/assets/10728166/54138e38-2c61-4c74-88db-71b1ca6b51ac)

**Comment by danielFesenmeyer** [comment]
Hi @Nickk4.

Seems that you are facing the issue described here: https://github.com/keycloak/keycloak/issues/19913

The best workaround seems to be this one here: https://github.com/keycloak/keycloak/discussions/8549#discussioncomment-2915578

**Comment by Nickk4** [comment]
Thanks @danielFesenmeyer, I've tried the script, which generated:

`INSERT INTO credential (id, type, user_id, secret_data, credential_data, priority) VALUES ('f0517703-***-2d85734d6fb2', 'password', 'a39776b5-***-ad20a9fccae9', '{"value": "4QVveEoBwm***5nfK75dY462+A==", "salt": "MAs***ZlxwSNw=="}', '{"hashIterations": 27500, "algorithm": "pbkdf2-sha256"}', 10);`

I ran this in my Postgres database, which was successful. However, after that, I was still unable to log in with the password I used for the script. Any ideas are appreciated.

P.S. Another method I've tried, which also didn't work, was to delete all records in `USER_ENTITY`, `CREDENTIAL` and `USER_ROLE_MAPPING`. I hoped this would then allow you to create a new admin user for the master realm, but that wasn't the case.

**Reply by danielFesenmeyer** [reply]
Have you cleared the cache or restarted after changing the password?

**Reply by Nickk4** [reply]
I'm not sure how to do that. I ran `KC_DB=postgres ./kc.sh start --hostname auth.mysite.com` after changing the password, but that keeps giving another error: 

<a href="https://ibb.co/CP78nT0"><img src="https://i.ibb.co/CP78nT0/image.png" alt="image" border="0"></a>

**Reply by danielFesenmeyer** [reply]
seems like you should stop keycloak first, because it's still running (address already in use)

**Reply by Nickk4** [reply]
Since I'm not sure how to stop KeyCloak I instead deleted both pods I have for KeyCloak (I still couldn't log in with my new password). Should that also prevent caching from logging in with the new password not working?

**Reply by tspearconquest** [reply]
Deleting the pods should serve to effectively do the same thing as stopping and then starting keycloak. Just to answer the question. :)

Did you get this solved?

**Reply by ejbest** [reply]
Anyone else test this?
